peer/x/roundrobin/list.go

```diff
@@ -210,7 +210,7 @@ func (pl *List) removeFromUnavailablePeers(p peer.Peer) {
 
 // Choose selects the next available peer in the round robin
 func (pl *List) Choose(ctx context.Context, req *transport.Request) (peer.Peer, func(error), error) {
-	if !pl.IsRunning() {
+	if err := pl.once.WaitForStart(ctx); err != nil {
 		return nil, nil, peer.ErrPeerListNotStarted("RoundRobinList")
```

Comment:
This seems like the smallest necessary change to me. We certainly could propagate the WaitForStart DeadlineExceeded or ErrAlreadyStopped, or wrap them both. I would favor keeping the existing behavior or just propagating the WaitForStart error without wrapping.